### PR TITLE
adding angular2-chat-app example

### PR DESCRIPTION
**Pull request by zubair-io**
Sorry for having to create a new pull request for this, my fork some how got disconnected and changes where not showing up.
The changes are:
Updated to angular 2.0.0-rc.1
Removing typings
Moving Horizon to a service

---

This change is [Reviewable](https://reviewable.io/reviews/rethinkdb/horizon/436)

### CODE REVIEW

**Comment by jonathan-casarrubias** [comment]
How to use the Horizon client without importing it within the index.html, in NativeScript we do not have an index.html so, I tried to import but not being able yet... any idea?

**Comment by marshall007** [comment]
@jonathan-casarrubias are you using TypeScript? I think you should be able to do:

``` ts
import * as Horizon from '@horizon/client'; 
```

**Comment by zubair-io** [comment]
Also add  `"@horizon/client": "1.0.1"` to the package.json

**Comment by jonathan-casarrubias** [comment]
Yeah I already tried that, I also created a simple definition and added it to the references.d.ts.

But when I try to compile I get the following error

`app/app.component.ts(2,21): error TS2307: Cannot find module '@horizon/client'.``

so the steps

1.- Created a NativeScript 2 App with Angular2 and TypeScript
2.- Created the definition and added it to the definitions
3.- installed @horizon/client 1.0.1
4.- Imported it from the app component

``` js
import {Component} from "@angular/core";
import * as Horizon from '@horizon/client'; 

@Component({
    selector: "my-app",
    template: `
<StackLayout>
    <Label text="Hello Horizon" class="title"></Label>
</StackLayout>
`,
})
export class AppComponent {}
```

When I visit the @horizon/client folder within the node_modules path, I see there is a dist, lib, package.json and README, but no index at all, so I found the files section within the package.json that actually points to dist and lib, but it does not work when trying to import it.

I also tried, to import `@horizon/client/dist` and `@horizon/client/lib` but still getting the same error.

If I use other 3rd party library like moment I can actually import it by using `import * as moment from 'moment'` so I can assume it is something specific with this horizon client library.

It is sad because I'm writing a blog post and project for this week about Horizon and NativeScript but this simple issue have me blocked.

I will keep playing with it to see if I can get to make it work, thanks in advanced for any help

Cheers,
Jon

**Comment by zubair-io** [comment]
@jonathan-casarrubias Here are some examples of nativescript running with Horizon  https://discuss.horizon.io/t/nativescript-starters-demos/152

**Comment by jonathan-casarrubias** [comment]
omg I was so god d**\* close, I was trying var Horizon = require("@horizon/client/lib"); when you posted the answer, because in there I see the index.js, which by the way does not work.

Now pointing to "@horizon/client/dist/horizon" does compile, thanks a lot @helixten it would be super awesome to have this in the documentation.. I can help if needed.. Also I know TS comes in the next major release, so maybe is not even needed..

Cheers
Jon

**Comment by poznyakovskiy** [comment]
The example is not working correctly. There's a bug in chat.component.ts, line 82: should be
`.sendChat(text)`

**Comment by zubair-io** [comment]
Thanks @poznyakovskiy fixed

File: examples/angular2-chat-app/dist/index.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<html>
+
+<head>
+  <title>Angular 2 QuickStart</title>
```
Comment (marcusreese) [inline examples/angular2-chat-app/dist/index.html:4]: Might want to change the title so it's no longer 'Angular 2 QuickStart'. And add <!doctype html> to the top?

**Comment by deadcee** [comment]
@zubair-io  Is it ok for me to help with anything on this?

**Comment by deadcee** [comment]
I have looked through this and it all appears to work perfectly apart from the last change suggested by @marcusreese so should be merged in my opinion.

**Comment by shantanubhadoria** [comment]
I can verify that this works on my local horizon setup. However I have to ask something possibly unrelated. Borrowing from the way it works on Meteor project is it possible to transpile, aggregate, minify and possibly uglify all the .ts on server side and keep the individual files in a folder not directly accessible by the client like imports/ ?
Thanks

**Comment by deontologician** [comment]
@shantanubhadoria This is why `hz serve` only serves files in the `dist/` directory. You can put your source files wherever you'd like and horizon won't serve them

**Comment by shantanubhadoria** [comment]
@deontologician Thanks for answering. How would I go about transpiling and aggregating .ts on the server side instead of loading individual components seperately on client with imports?
Thanks

**Comment by zubair-io** [comment]
@shantanubhadoria I think you are looking for a bundler. Try adding  webpack or systemjs-bundler

**Pull request closed without merging**